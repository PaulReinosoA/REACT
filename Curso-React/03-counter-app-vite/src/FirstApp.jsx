import PropTypes from 'prop-types';
//import {Fragment} from 'react'

//Un nodo padre es un Fragmento o <>
//pues varios elementos html necesitan agrupadores 
//<> es un fragmento sin neceidad de la importacion

//<!--debe ser una exprecion y no un objeto
const newMessaje =[1,2,3,4,5,6,7];

const newMessajeObj = {
    message:'Hola mundo',
    title:'Fernando'
};

const getMessage = ()=>{
    return 'Saludo Paul soy una funcion()!!!'
}

const getResult = (a,b)=>{
    return a * b;
}

//Las funciones no van dentro del functional component 
//a no ser que sean casoso como eventos de botones etc,
//o funciones que necesiten estar dentor del functional component
/*
export const FirstApp = ( props ) => {
    //console.log(props)//son un canal de comunicacion entre el Comp. padre del app//desestructura simepre
   
    return (
        <>
            <h1>FIRST APP</h1>
            <h1>{ props.title }</h1>  
            <h1>{ props.subtitle }</h1>  
            <h1>{ 2+5 }</h1>             
            <code>{ JSON.stringify(newMessajeObj)  }</code>
            <h1>{ newMessaje }</h1>
            <p>{ getResult(4,3) }</p>
            <p>{ getMessage() }</p>
            <p>Soy el subtitulo</p>
        </>
    )
}*/


export const FirstApp = ( {title,subtitle,name } ) => {
    //console.log(props)//son un canal de comunicacion entre el Comp. padre del app//desestructura simepre
   
    return (
        <>
            <h1>FIRST APP</h1>
            <h1>{ title }</h1>  
            <h1>{ subtitle }</h1> 
            <p>{ name }</p>

        </>
    )
}

FirstApp.propTypes = {
    title:PropTypes.string.isRequired,
    subtitle:PropTypes.number.isRequired,
};

FirstApp.defaultProps = {
    
    //title :'No hay titulo',
    subtitle : 0,    
    name : 'Paul Reinoso'
};




