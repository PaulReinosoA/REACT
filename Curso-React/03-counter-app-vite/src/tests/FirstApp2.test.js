import { render, screen, getAllByText } from "@testing-library/react";
import { FirstApp } from "../FirstApp";
import { toMatchSnapshot } from "jest-snapshot";


describe('Pruebas sobre  <FirstApp />', () => {

    const title = 'Hola Soy, Goku'
    
    test('Debe comparar el snapshot ', () => {
      
        const {container} = render(<FirstApp title = { title } />);
        expect(container).toMatchSnapshot();
    })
   
    test('Bede mostar el mensaje "Hola soy, Goku" ', () => {

        render(<FirstApp title = { title } />);
        expect(screen.getByText(title)).toBeTruthy(); //El screen -->>> objeto que estamos renderizando 
        screen.debug(); //ultima version actualizadad despues de los cambios del DOM

    });

    test('debe mostrar el titulo en un H1 ', () => {
        
        render(<FirstApp title = { title } />);
        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);  //el roll son elementos html

    })
    
    test('Ddebe de mostarr el subtitulo enviado por props ', () => {
        
        render(
            <FirstApp 
                title = { title } 
                subtitle = { subtitle } 
            />
        );

        expect(screen.getAllByText(subtitle).length).toBe(2);
    })
    
});
