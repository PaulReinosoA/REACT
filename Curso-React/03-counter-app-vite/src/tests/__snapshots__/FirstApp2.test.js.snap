// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre  <FirstApp /> Debe comparar el snapshot  1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola Soy, Goku
  </h1>
  <p>
    0
  </p>
  <p>
    0
  </p>
  <p>
    Paul Reinoso
  </p>
</div>
`;
